src/store/store.ts

```diff
@@ -308,9 +318,16 @@ export const useStore = defineStore('store', {
       return this.stateModel.tombstone.businessId
     },
 
-    /** The original legal name. */
+    /** The original legal name (or alternate name if this is a firm). */
     getOriginalLegalName (): string {
-      return this.getEntitySnapshot?.businessInfo?.legalName
+      if (!GetFeatureFlag('enable-legal-name-fix')) return this.getOriginalBusinessInfo?.legalName
+      if (this.isFirm) {
+        // return the alternate name, if it exists
+        const alternateNames = this.getOriginalBusinessInfo?.alternateNames || []
+        const alternateName = alternateNames.find(x => x.identifier === this.getBusinessId)
+        return alternateName?.name || 'Unknown'
```

Comment:
Returning Unknown instead of null, here, is probably OK because firms cannot be numbered.

Reply:
(I believe code checks if `getOriginalLegalName` is falsy, and if so then it displays the numbered legal name.

Reply:
Yup, makes sense. I also made sure to test. It should be OK.